lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2283,8 +2297,17 @@ private static function get_block_nodes( $theme_json, $selectors = array() ) {
 				'selectors'  => $feature_selectors,
 				'duotone'    => $duotone_selector,
 				'variations' => $variation_selectors,
+				// 'state_selectors' => $selectors[ $name ]['.current-item'],
+			);
+
+			$nodes[] = array(
+				'name' => 'core/navigation',
+				'path' => array( 'styles', 'blocks', 'core/navigation', '.current-item' ),
```

Comment:
This is a hack. 

We can either:

- have the code that processes the block `$node` understand how to handle styles for `.current-item`.
- add a new loop in this location which processes it (a bit like how pseudo elements are handled below).

I prefer the former approach as it's more declarative.